File: springboot-azure/spring-boot-azure-cosmos-db-sql-api-crud/src/main/java/com/knf/dev/demo/exception/GlobalExceptionHandler.java
package com.knf.dev.demo.exception;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.context.request.WebRequest;

@ControllerAdvice
public class GlobalExceptionHandler {

	@ExceptionHandler(UnknownError.class)
	public ResponseEntity<String> internalServerError(Exception ex, WebRequest request) {

		return new ResponseEntity<>(ex.getMessage(), HttpStatus.INTERNAL_SERVER_ERROR);
	}

	@ExceptionHandler(UserNotFound.class)
	public ResponseEntity<String> userNotFound(Exception ex, WebRequest request) {

		return new ResponseEntity<>(ex.getMessage(), HttpStatus.NOT_FOUND);
	}
}

File: kotlin-spring-main/kotlin-springboot-mybatis-crud-example/src/main/kotlin/com/knf/dev/demo/exception/GlobalExceptionHandler.kt
package com.knf.dev.demo.exception

import org.springframework.web.bind.annotation.ControllerAdvice
import com.knf.dev.demo.exception.UserIdNotFoundException
import org.springframework.web.context.request.WebRequest
import org.springframework.http.ResponseEntity
import java.time.LocalDateTime
import org.springframework.http.HttpStatus
import com.knf.dev.demo.exception.UserIdAlreadyExistException
import org.springframework.web.bind.annotation.ExceptionHandler
import java.lang.Exception

@ControllerAdvice
class GlobalExceptionHandler {
    @ExceptionHandler(UserIdNotFoundException::class)
    fun globalExceptionHandler(ex: Exception, request: WebRequest?):
            ResponseEntity<CustomErrorResponse> {
        val errors = CustomErrorResponse()
        errors.timestamp = LocalDateTime.now()
        errors.error = ex.message
        errors.status = HttpStatus.NOT_FOUND.value()
        return ResponseEntity<CustomErrorResponse>(errors, HttpStatus.NOT_FOUND)
    }

    @ExceptionHandler(UserIdAlreadyExistException::class)
    fun globalExceptionHandler2(ex: Exception, request: WebRequest?):
            ResponseEntity<CustomErrorResponse> {
        val errors = CustomErrorResponse()
        errors.timestamp = LocalDateTime.now()
        errors.error = ex.message
        errors.status = HttpStatus.INTERNAL_SERVER_ERROR.value()
        return ResponseEntity<CustomErrorResponse>(errors,
            HttpStatus.INTERNAL_SERVER_ERROR)
    }
}